Dockerfile

```diff
@@ -25,9 +25,9 @@ RUN go mod download
 COPY . .
 RUN go build -v -o /usr/local/bin/kpt -ldflags="-s -w -X github.com/GoogleContainerTools/kpt/run.version=$KPT_VERSION" ./
 
-FROM alpine:3.11
+FROM gcr.io/google.com/cloudsdktool/cloud-sdk:alpine
```

Comment:
We have started standardizing on specific versions of alpine for functions, we should follow the same approach here. @mengqiy 

@mortent Can you also elaborate (probably add to PR description) why you're making this change given the binary bloat to 200MB? Have you looked at re

Reply:
In kpt functions, we are using alpine:3.12 for go-based and node:15.3-alpine3.12 for ts-based.
For Googlers, to see the complete description: https://docs.google.com/document/d/1q22he3sQF-w2Jn78YprG6orxx0Avy3N105QIqe51Vso/edit#heading=h.v1kcq2swzui1

https://github.com/GoogleCloudPlatform/cloud-sdk-docker/blob/master/alpine/Dockerfile is how gcr.io/google.com/cloudsdktool/cloud-sdk:alpine is built. It seems it installs a lot of other stuffs besides gcloud. If we can avoid install unnecessary stuffs, we make the size smaller.

Reply:
So using `kpt live` with the kpt image requires gcloud for auth. I expect that the stuff installed into the cloud-sdk image is there for a reason, so I think it is safer for us to use the validated image for this instead of rolling our own, even if the cloud-sdk image uses a different version of alpine.

I suggest we just create a separate image for kpt that installs gcloud. That will allow users who don't need the `kpt live` functionality to use the small image, but we provide an image for users how want to use `kpt live`.